runtime/codert_vm/cnathelp.cpp

```diff
@@ -2425,6 +2423,40 @@ old_slow_jitInduceOSRAtCurrentPC(J9VMThread *currentThread)
 	return newPC;
 }
 
+void* J9FASTCALL
+old_slow_jitInduceOSRAtCurrentPCAndRecompile(J9VMThread *currentThread)
+{
+	OLD_JIT_HELPER_PROLOGUE(0);
+	void *oldPC = buildJITResolveFrame(currentThread, J9_SSF_JIT_RESOLVE_INDUCE_OSR, parmCount);
+
+	J9StackWalkState walkState;
+	/* The stack currently has a resolve frame on top, followed by the JIT frame we
+	 * wish to OSR.  Walk the stack down to the JIT frame to gather the data required
+	 * for decompilation.
+	 */
+	J9JavaVM *vm = currentThread->javaVM;
+	J9JITConfig *jitConfig = vm->jitConfig;
+	walkState.walkThread = currentThread;
+	walkState.maxFrames = 2;
+	walkState.flags = J9_STACKWALK_SKIP_INLINES | J9_STACKWALK_MAINTAIN_REGISTER_MAP | J9_STACKWALK_COUNT_SPECIFIED;
+	currentThread->javaVM->walkStackFrames(currentThread, &walkState);
+	J9Method * method = walkState.method;
+	void *jitPC = walkState.pc;
+	J9JITExceptionTable *metaData = jitConfig->jitGetExceptionTableFromPC(currentThread, (UDATA)jitPC);
+	void *oldJITStartAddr = (void *) metaData->startPC;
+	jitConfig->retranslateWithPreparationForMethodRedefinition(jitConfig, currentThread, method, oldJITStartAddr);
```

Comment:
Strictly speaking, you don't need to go via the jitConfig here if a prototype is available - this file is in the JIT library.

Reply:
May as well leave this this way for now.